src/components/navs/AppNav/AppNavNetworkSelect.vue

```diff
@@ -94,23 +93,19 @@ onMounted(async () => {
 });
 
 // WATCHERS
-watchEffect(() => {
-  if (
-    chainId.value &&
-    previousNetwork.value &&
-    previousNetwork.value !== chainId.value
-  ) {
-    const newNetwork = allNetworks.value.find(
-      n => Number(n.key) === chainId.value
-    );
-    if (newNetwork) {
-      document.body.style.display = 'none';
-      localStorage.setItem('networkId', chainId.value.toString());
-      setWindowLocation(getNetworkChangeUrl(newNetwork));
+watch(
+  () => chainId.value,
```

Comment:
Can this not just be:
```suggestion
  chainId,
```